### ISSUE TITLE
Interface for querying rendermime safe/sanitization

### ISSUE THREAD

**Issue by vidartf** [jupyterlab/jupyterlab#1824]
It would be nice to have some more functions on `RenderMime` to inquire about its capabilities for rendering.

### Use case
I need to determine whether the `trusted` value of an output model will affect the rendering of any of its mimetypes (i.e. the mimetypes that are present in the bundle and that the renderer supports). Something like this:

```ts
let isTrustSignificant = false;
for (let mimetype of intersection(output.mimetypes, rendermime.supportedMimetypes)) {
  if (!rendermime.isSafe(mimetype)) {
    isTrustSignificant = true;
    break;
  } else if (rendermime.isSanitizable(mimetype)) {
    // Here, we might want to check whether sanitization will actually
    // affect output, but that might be expensive...
    isTrustSignificant = true;
    break;
  }
}
```

Expected results for the code above:
 - `text/html` -> true
 - `image/png` -> false
 - `image/svg+xml` -> true
 - `application/javascript` -> true
 - `text/plain` -> false

**Comment by vidartf** [comment]
It feels like this is _almost_ possible by `rendermime.getRenderer()` + `renderer.canRender()`, but I'm not sure what would need to be passed as `sanitizer` to `canRender()`.

**Comment by blink1073** [comment]
If we made `sanitizer` optional again, the renderer would have to take that into account in `canRender()`, but that would only apply to renderers that care about sanitization, versus forcing all renderers to have an `isSanitizable` interface.  Sounds good to me.

**Comment by vidartf** [comment]
It seems like none of the actual renderers look at the `sanitizer` argument to `canRender` ?

**Comment by vidartf** [comment]
Maybe each renderer could have a `readonly sanitizes: boolean` ? Or possibly a method `wouldSanitize({mimeType: string, model: IMimeModel}): boolean` for more extensibility in the future.

**Comment by blink1073** [comment]
The current behavior is that they *always* get a sanitizer, because we have a default sanitizer if none is given.  The `wouldSanitize()` method sounds fair.

**Comment by vidartf** [comment]
Maybe the sanitizer on `RenderMime` should be exposed as (readonly) public then? Because you would want to pass its sanitizer to `canRender` etc.

**Comment by blink1073** [comment]
Yep, also sounds good.

**Comment by blink1073** [comment]
Closed by #1836.
